test/conftest.py

```diff
@@ -265,20 +250,27 @@ def retriever_with_docs(request, document_store_with_docs):
     return get_retriever(request.param, document_store_with_docs)
 
 
-def get_document_store(document_store_type, faiss_document_store, inmemory_document_store):
+def get_document_store(document_store_type):
     if document_store_type == "sql":
         if os.path.exists("haystack_test.db"):
             os.remove("haystack_test.db")
         document_store = SQLDocumentStore(url="sqlite:///haystack_test.db")
     elif document_store_type == "memory":
-        document_store = inmemory_document_store
+        document_store = InMemoryDocumentStore(return_embedding=False)
     elif document_store_type == "elasticsearch":
         # make sure we start from a fresh index
         client = Elasticsearch()
         client.indices.delete(index='haystack_test*', ignore=[404])
         document_store = ElasticsearchDocumentStore(index="haystack_test", return_embedding=False)
     elif document_store_type == "faiss":
-        document_store = faiss_document_store
+        if os.path.exists("haystack_test_faiss.db"):
+            os.remove("haystack_test_faiss.db")
+        document_store = FAISSDocumentStore(
+            sql_url="sqlite:///haystack_test_faiss.db",
+            return_embedding=False
+        )
+        document_store.faiss_index.reset()
```

Comment:
This will create new index hence old test run memory won't be cleaned.
Better to keep `faiss_document_store` separately and use that here.